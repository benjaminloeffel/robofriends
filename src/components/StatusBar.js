/** @jsx jsx */
import React from "react"; // eslint-disable-line no-unused-vars
import { jsx } from "theme-ui";
import { Styled } from "theme-ui";
import { Box } from "@theme-ui/components";
import PropTypes from "prop-types";

const StatusBar = ({ loading, error, resultLength }) => {
  return (
    <Box sx={{ bg: "text", padding: "16px", textAlign: "center" }}>
      <Styled.p sx={{ color: "background" }}>
        {loading ? (
          <>
            Please wait while we're trying to load your data.
            <span role="img" aria-label="Loading">
              ‚è≥
            </span>
          </>
        ) : error ? (
          <>
            Oh no! We've encountered an error!
            <span role="img" aria-label="Screaming">
              üò±
            </span>
          </>
        ) : resultLength > 0 ? (
          <>
            We have found {resultLength} robots matching your search.{" "}
            <span role="img" aria-label="Robot">
              ü§ñ
            </span>
          </>
        ) : (
          <>
            Sorry, we could not find any robots matching your search.
            <span role="img" aria-label="Sad">
              üôÅ
            </span>
          </>
        )}
      </Styled.p>
    </Box>
  );
};

StatusBar.propTypes = {
  loading: PropTypes.bool.isRequired,
  error: PropTypes.object.isRequired,
  resultLength: PropTypes.number.isRequired
};

export default StatusBar;
