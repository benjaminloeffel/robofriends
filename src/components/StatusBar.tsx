/** @jsx jsx */
import React from "react"; // eslint-disable-line no-unused-vars
import { jsx } from "theme-ui";
import { Styled } from "theme-ui";
import { Box } from "@theme-ui/components";
import PropTypes, { InferProps } from "prop-types";

const StatusBar: React.FC<Props> = ({ loading, error, resultLength }) => {
  return (
    <Box sx={{ bg: "text", padding: "16px", textAlign: "center" }}>
      <Styled.p sx={{ color: "background" }}>
        {loading ? (
          <React.Fragment>
            Please wait while we're trying to load your data.
            <span role="img" aria-label="Loading">
              ‚è≥
            </span>
          </React.Fragment>
        ) : error.name ? (
          <React.Fragment>
            Oh no we've encountered {error.name}: {error.message}
            <span role="img" aria-label="Screaming">
              üò±
            </span>
          </React.Fragment>
        ) : resultLength > 0 ? (
          <React.Fragment>
            We have found {resultLength} robots matching your search.{" "}
            <span role="img" aria-label="Robot">
              ü§ñ
            </span>
          </React.Fragment>
        ) : (
          <React.Fragment>
            Sorry, we could not find any robots matching your search.
            <span role="img" aria-label="Sad">
              üôÅ
            </span>
          </React.Fragment>
        )}
      </Styled.p>
    </Box>
  );
};

const userPropTypes = {
  loading: PropTypes.bool.isRequired,
  error: PropTypes.shape({
    name: PropTypes.string,
    message: PropTypes.string
  }).isRequired,
  resultLength: PropTypes.number.isRequired
};

const userDefaultProps = {};

type InferPropTypes<
  PropTypes,
  DefaultProps = {},
  Props = InferProps<PropTypes>
> = {
  [Key in keyof Props]: Key extends keyof DefaultProps
    ? Props[Key] | DefaultProps[Key]
    : Props[Key];
};

type Props = InferPropTypes<typeof userPropTypes, typeof userDefaultProps>;

export default StatusBar;
