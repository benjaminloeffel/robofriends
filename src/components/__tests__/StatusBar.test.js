import React from "react";
import "../../setupTests";
import { shallow } from "enzyme";
import StatusBar from "../StatusBar";

const mockProps = {
  loading: false,
  error: { status: false, message: "" },
  resultLength: 1
};

const wrapper = shallow(<StatusBar {...mockProps} />);

describe("StatusBar", () => {
  it("renders the StatusBar component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders the loading message when 'loading' prop is true", () => {
    const mockProps2 = {
      loading: true,
      error: { status: false, message: "" },
      resultLength: 1
    };
    const wrapper2 = shallow(<StatusBar {...mockProps2} />);

    expect(wrapper2).toMatchSnapshot();
  });

  it("renders the error message when 'error' prop is true", () => {
    const mockProps3 = {
      loading: false,
      error: {
        status: true,
        message: "Sorry, we could not find any robots matching your search."
      },
      resultLength: 0
    };
    const wrapper3 = shallow(<StatusBar {...mockProps3} />);

    expect(wrapper3).toMatchSnapshot();
  });

  it("renders the error message when 'error' prop is true", () => {
    const mockProps4 = {
      loading: false,
      error: {
        status: false,
        message: "Sorry, we could not find any robots matching your search."
      },
      resultLength: 0
    };
    const wrapper4 = shallow(<StatusBar {...mockProps4} />);

    expect(wrapper4).toMatchSnapshot();
  });
});
