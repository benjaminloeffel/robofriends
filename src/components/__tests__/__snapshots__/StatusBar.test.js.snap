// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar renders the StatusBar component 1`] = `ShallowWrapper {}`;

exports[`StatusBar renders the error message when 'error' prop is true 1`] = `ShallowWrapper {}`;

exports[`StatusBar renders the error message when 'error' prop is true 2`] = `ShallowWrapper {}`;

exports[`StatusBar renders the loading message when 'loading' prop is true 1`] = `ShallowWrapper {}`;
